GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -637,9 +660,38 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
 
                     break;
 
+                case FileSystemTask.OperationType.OnFilePreDelete:

```

Comment:
Maybe add a comment in the `PreDelete` and `Delete` cases and mention that currently the code assumes that `PreDelete` and `Delete` will never both be queued up?

If in the future they are both queued, we'll want some sort of `GVFSPlatform` check (e.g. `SupportsPostDeleteNotification`) to avoid doing this work twice.